GoogleChromeLabs/ps-analysis-tool
# Feature: Add warning and collect details of URL errors in sitemap

## Description
This PR aims to collect errors on URL which occur while analysing the sitemap.

## Relevant Technical Choices
- The CLI catches critical errors when sitemap analysis is running.
- These errors are accumulated, processed and then these URLs are displayed on the CLI-Dashboard.

## Testing Instructions
- Clone this branch.
- In the terminal, run `npm i` and run `npm run build:cli:dashboard`.
- Use a CSV or a sitemap where there are URLs with errors.
- After analysis of the sitemap, open the dashboard, you'll notice those URLs listed under URLs with errors. 

**Example:**
CSV File: [psat-csv-test.csv](https://github.com/user-attachments/files/16903761/psat-csv-test.csv)
Run `npm run cli -- -f ~/Downloads/psat-csv-test.csv`

## Additional Information:



## Screenshot/Screencast



---

## Checklist






Fixes #818
